### DISCUSSION TITLE
After switching to Turbopack i get crypto.getRandomValues() not supported.

### DISCUSSION THREAD

**Discussion by kapitalistn** [nextauthjs/next-auth#7613] [Help] [answered]
### Question 💬

When I enable Turbopack i get following error: 

[next-auth][error][JWT_SESSION_ERROR]
https://next-auth.js.org/errors#jwt_session_error crypto.getRandomValues() not supported. See https://github.com/uuidjs/uuid#getrandomvalues-not-supported {
  message: 'crypto.getRandomValues() not supported. See https://github.com/uuidjs/uuid#getrandomvalues-not-supported',
  stack: 'Error: crypto.getRandomValues() not supported. See https://github.com/uuidjs/uuid#getrandomvalues-not-supported\n' +
    '    at rng (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\node_modules_34ddeb._.js:235203:19)\n' +
    '    at v4 (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\node_modules_34ddeb._.js:235222:191)\n' +
    '    at Object.encode (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\node_modules_34ddeb._.js:215033:76)\n' +
    '    at async Object.session (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\node_modules_34ddeb._.js:215940:30)\n' +
    '    at async AuthHandler (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\node_modules_34ddeb._.js:217578:37)\n' +
    '    at async Module.getServerSession (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\node_modules_34ddeb._.js:217906:21)\n' +
    '    at async getSession (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\app_actions_getCurrentUser.ts_e5e7e1._.js:17:12)\n' +
    '    at async Module.getCurrentUser (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\app_actions_getCurrentUser.ts_e5e7e1._.js:21:25)\n' +
    '    at async GET (C:\\akademia\\.next\\server\\app\\chunks\\ssr\\app_api_courses_route.ts_f91375._.js:49:25)\n' +
    '    at async C:\\akademia\\.next\\server\\app\\chunks\\ssr\\node_modules_34ddeb._.js:210465:37',
  name: 'Error'
}
warning - [parse]
  ... [2 paths] are hidden, run with --show-all to show them
event - 7 requests (GET /api/courses, ...) 1540ms
event - 62 requests (GET /turbopack-hmr, ...) 40ms
event - 7 requests (GET /api/courses, ...) 745ms

### How to reproduce ☕️

Install Next.js 13.4.2, NextAuth 4.22.1, bcryptjs 2.4.3 and enable turbopack in package.json.

Configure credentials sign in.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by godfrednanaowusu** [comment]
multiple issues regarding this have been converted to discussions but nothing seems to happen about it, its been about 3 weeks now if not more

**Reply by kapitalistn** [reply]
Maybe they're fully concentrated on Auth.js... 
Btw this is the only issue why I can't use Turbopack...

**Comment by godfrednanaowusu** [comment]
same here, only thing stopping me from using turbopack.

It looks like all efforts are now on auth.js, and that also looks a long way from being available, it's become so dicey.

I have been monitoring for almost a month now, and no single activity on this.
They are always very quick to convert the issues into a discussion and then leave them there to die it seems.

Also almost a year since app directory was released, there isn't any proper documentation for it, it's painful and pathetic.

**Comment by romikusumabakti** [comment, marked as answer]
I got it working by adding the following snippet to the `app\api\auth\[...nextauth]\route.ts`:

```ts
import crypto from "node:crypto";
global.crypto ??= crypto;
```

**Reply by kapitalistn** [reply]
Thanks, you're my hero now)

**Reply by christo9090** [reply]
This does work for me as well but I get the following typescript error. 
`Type 'typeof import("node:crypto")' is not assignable to type 'Crypto'.
  The types returned by 'randomUUID(...)' are incompatible between these types.
    Type 'string' is not assignable to type '`${string}-${string}-${string}-${string}-${string}`'.ts(2322)`

**Comment by godfrednanaowusu** [comment]
It worked
